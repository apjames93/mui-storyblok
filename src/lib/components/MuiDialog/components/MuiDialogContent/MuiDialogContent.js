import React, { createElement } from 'react';
import PropTypes from 'prop-types';
import DialogContent from '@material-ui/core/DialogContent';
import Storyblok from '../../../../utils/Storyblok';
import BlokForm from '../../../BlokForm/BlokForm';
import MuiDialogContentTypography from './components/MuiDialogContentTypography/MuiDialogContentTypography';
import MuiExpansionPanel from '../../../MuiExpansionPanel/MuiExpansionPanel';

const MuiDialogContent = ({
  dividers,
  content,
  rootClass,
}) => {
  const styles = Storyblok.arrayToMuiStyles(rootClass);
  const components = {
    MuiDialogContentTypography,
    MuiExpansionPanel,
    BlokForm,
  };

  return (
    <DialogContent
      className={styles.root}
      dividers={dividers}
    >
      {content.map((item, index) => createElement(
        components[item.component],
        Object.assign(item, { key: index }),
      ))}
    </DialogContent>
  );
};

export default MuiDialogContent;

MuiDialogContent.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Override or extend the styles applied to the component.
   * */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * mui prop: true | false
   * Display the top and bottom dividers.
   * */
  dividers: PropTypes.bool,
  /** MuiTypography MuiExpansionPanel, BlokForm */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiDialogContent.defaultProps = {
  dividers: false,
  rootClass: [],
};
