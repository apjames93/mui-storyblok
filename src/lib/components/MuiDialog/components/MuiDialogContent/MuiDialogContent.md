##### MuiDialogContent has the ability to render custom components.