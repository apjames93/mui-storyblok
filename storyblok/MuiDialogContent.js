module.exports = {
  "name": "MuiDialogContent",
  "display_name": null,
  "updated_at": "2020-07-20T17:00:14.628Z",
  "schema": {
    "dividers": {
      "type": "boolean"
    },
    "rootClass": {
      "type": "options"
    },
    "content": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "MuiTypography",
        "MuiGrid",
        "BlokForm",
        "MuiAccordion"
      ]
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiDialogContent",
  "component_group_uuid": null
}