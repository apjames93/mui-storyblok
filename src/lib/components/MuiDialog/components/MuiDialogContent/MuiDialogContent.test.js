import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiDialogContent from './MuiDialogContent';

function setup() {
  const props = {
    content: [{
      component: 'MuiDialogContentTypography',
      content: [{
        component: 'MuiText',
        text: 'text',
      }],
    }],
  };
  const comp = shallow(<MuiDialogContent {...props} />);

  return { comp, props };
}

describe('<MuiDialogContent />', () => {
  it('renders MuiDialogContent', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });


  test('snapshot', () => {
    const { props } = setup();

    const tree = renderer.create((
      <MemoryRouter>
        <MuiDialogContent {...props} />
      </MemoryRouter>
    ));
    expect(tree).toMatchSnapshot();
  });
});
