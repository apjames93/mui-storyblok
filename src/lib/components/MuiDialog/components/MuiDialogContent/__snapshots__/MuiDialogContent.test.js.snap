// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiDialogContent /> snapshot 1`] = `
<div
  className="MuiDialogContent-root makeStyles-root-3"
>
  <div
    className="MuiTypography-root makeStyles-root-4 MuiTypography-body1"
  >
    <span
      className="MuiTypography-root makeStyles-root-5 MuiTypography-body1"
    >
       undefined 
    </span>
  </div>
</div>
`;
